rotkehlchen/icons.py

```diff
@@ -127,7 +137,7 @@ def query_coingecko_for_icon(self, asset: Asset, coingecko_id: str) -> bool:
         # we only keep delisted asset coingecko mappings since historical prices
         # can still be queried.
         if asset.identifier in DELISTED_ASSETS:
-            self.failed_asset_ids.add(asset.identifier)
+            self.failed_asset_ids.add(asset.identifier, asset.identifier)
```

Comment:
This is useless. Change it also everywhere else you use it. you are making a cache with double the needed size. You are caching the asset identifier both as the key and the value in the cache. For the value you don't need anything. Just a  boolean would be enough.

In fact this cache you are using is a key-value mapping cache. You don't need this. You need a set LRU cache.